src/Test/FixturesTrait.php

```diff
@@ -45,15 +45,15 @@ protected function getContainer(): ContainerInterface
             $options = [
                 'environment' => $environment,
             ];
-            $kernel = $this->createKernel($options);
-            $kernel->boot();
-
-            $container = $kernel->getContainer();
-            if ($container->has('test.service_container')) {
-                $this->containers[$environment] = $container->get('test.service_container');
-            } else {
-                $this->containers[$environment] = $container;
+
+            // Check that the kernel has not been booted separately (eg. with static::createClient())
+            $kernel = static::$kernel;
+            if (null === $kernel) {
```

Comment:
you don't need a $kernel variable? `if (null === static::$kernel) { $this->bootKernel($options); }`